lib/rules/prefer-const.js

```diff
@@ -170,7 +170,12 @@ function getIdentifierIfShouldBeConst(variable, ignoreReadBeforeAssign) {
                     const elements = leftNode.elements;
 
                     hasOuterVariables = elements
-                        .map(element => element.name)
```

Comment:
`element && element.name`? or `.filter(Boolean).map(element => element.name)`?

Reply:
@ljharb :man_facepalming: yep makes sense!

Reply:
or it can be in one iteration, like:
```js
elements.some(element => element && element.name && isOuterVariableInDestructing(element.name, variable.scope) );
```